unstructured/chunking/title.py

```diff
@@ -9,11 +10,80 @@
     Element,
     ElementMetadata,
     Table,
+    TableChunk,
     Text,
     Title,
 )
 
 
+def chunk_by_characters(
+    elements: List[Element],
+    num_characters: Optional[int] = 200,
+) -> List[Element]:
+    """Uses num_characters to return elements within a document in specific sizes.
+    Currently only applies to Table elements.
+     Parameters
+    ----------
+    elements
+        A list of unstructured elements. Usually the output of a partition functions.
+    num_characters
+        Splits elements text field so that a section matches the desired length."""
+
+    chunked_elements: List[Element] = []
+
+    for element in elements:
+        if isinstance(element, Table):
+            char_len = len(element.text)
+            html_table = element.to_dict().get("text_as_html")
+            if html_table:
+                char_len = len(html_table)
+            if char_len > num_characters:  # type: ignore
+                chunked_elements.extend(chunk_table_element(element, num_characters))
+            else:
+                chunked_elements.append(element)
+            # create TableChunk element if text is more than num chars
+        else:
+            # TODO (Amanda): extend to other element types
+            chunked_elements.append(element)
+
+    return chunked_elements
+
+
+def chunk_table_element(
+    element: Table,
+    num_characters: Optional[int] = 200,
+) -> List[TableChunk]:
+    chunks = []
+    text = element.text
+    text_as_html = element.text_as_html if hasattr(element, "text_as_html") else None
+    i = 0
+    metadata = element.metadata
+    while text or text_as_html:
+        text_chunk = text[:num_characters]
+        table_chunk = TableChunk(
+            text=text_chunk,
+            metadata=copy.copy(metadata),
+        )
+        if text_as_html:
+            text_as_html_chunk = text_as_html[:num_characters]
+            table_chunk.metadata.text_as_html = text_as_html_chunk
+            # Remove the processed chunk from text_as_html
+            text_as_html = text_as_html[num_characters:]
```

Comment:
so there is really no correlation between the text chunk and the `text_as_html` chunk since the html will have html formatting characters so it will be different from text
additionally what does this mean to this chunk? This chunk would still contain `2 * num_character` of text if we got both html and text. And this means the total number of characters in this chunk is twice that of other chunks without html. What would this mean for RAG downstream (main usage for chunks?)

Reply:
That's correct - this chunk will still contain text and text_as_html params, both potentially at the given char # input.

My understanding is that the downstream workflow understands tables better in their HTML form, so I'm guessing if someone is chunking tables they're sending the text_as_html elements and not the text, or separately? But it's definitely important to provide the chunking for the text_as_html param if the element has it after partitioning.